tests/snmp/test_snmp_loopback.py

```diff
@@ -66,8 +70,11 @@ def test_snmp_loopback(duthosts, enum_rand_one_per_hwsku_frontend_hostname,
             continue
         result = get_snmp_output(loip, duthost, nbr, creds_all_duts)
         assert result is not None, 'No result from snmpget'
-        assert len(result['stdout_lines']) > 0, 'No result from snmpget'
-        assert "SONiC Software Version" in result['stdout_lines'][0][0],\
-            "Sysdescr not found in SNMP result from IP {}".format(ip)
-        assert snmp_facts['ansible_sysdescr'] in result['stdout_lines'][0][
-            0], "Sysdescr from IP{} not matching with result from Mgmt IPv4.".format(ip)
+        assert len(result[u'stdout_lines']) > 0, 'No result from snmpget'
+
+        if isinstance(nbr['host'], EosHost):
+            assert "SONiC Software Version" in result[u'stdout_lines'][0][0], "Sysdescr not found in SNMP result from IP {}".format(ip)
+            assert snmp_facts['ansible_sysdescr'] in result[u'stdout_lines'][0][0], "Sysdescr from IP{} not matching with result from Mgmt IPv4.".format(ip)
+        else:
+            assert "SONiC Software Version" in result[u'stdout_lines'][0], "Sysdescr not found in SNMP result from IP {}".format(ip)
```

Comment:
the assert is checking for the result.
Could modify get_snmp_output() to get the result in a uniform format so that the test function need not check for the type of nbr host.